recipe-server/contract-tests/test_api.py

```diff
@@ -117,6 +117,27 @@ def test_recipe_signatures(conf, requests_session):
         signing.verify_x5u(url)
 
 
+def test_action_signatures(conf, requests_session):
```

Comment:
Contract tests are designed to validate existing servers, not as unit tests. As such there isn't anyway to use them to force a invalid hash or signature.